src/components/pages/Triggers/TriggerDetails/TriggerDetails.tsx

```diff
@@ -0,0 +1,71 @@
+import {useContext, useEffect, useState} from 'react';
+import {Helmet} from 'react-helmet';
+
+import {Tabs} from 'antd';
+
+import {ConfigContext, DashboardContext, MainContext} from '@contexts';
+
+import useLocation from '@hooks/useLocation';
+
+import {useGetTriggerByIdQuery, useGetTriggersKeyMapQuery} from '@services/triggers';
+
+import {useShallowGlobalStore} from '@store/GlobalStore';
+
+import {safeRefetch} from '@utils/fetchUtils';
+
+import {StyledContainer, StyledPageHeader} from './TriggerDetails.styled';
+import TriggerSettings from './TriggerSettings';
+
+const TriggerDetails = () => {
+  const {isClusterAvailable} = useContext(MainContext);
+  const {location, navigate} = useContext(DashboardContext);
+  const {pageTitle} = useContext(ConfigContext);
+
+  const {currentTrigger, setCurrentTrigger, setTriggersKeyMap} = useShallowGlobalStore(state => ({
+    currentTrigger: state.currentTrigger,
+    setCurrentTrigger: state.setCurrentTrigger,
+    setTriggersKeyMap: state.setTriggersKeyMap,
+  }));
+
+  const name = useLocation().lastPathSegment;
+
+  const [activeTabKey, setActiveTabKey] = useState('Settings');
+
+  const {data: triggerDetails, refetch} = useGetTriggerByIdQuery(name, {skip: !isClusterAvailable});
+  const {data: triggersKeyMap, refetch: refetchKeyMap} = useGetTriggersKeyMapQuery(null, {skip: !isClusterAvailable});
+
+  const isPageDisabled = !name;
+
+  useEffect(() => {
+    if (triggerDetails) {
+      setCurrentTrigger(triggerDetails);
+    }
+  }, [triggerDetails]);
+
+  useEffect(() => {
+    if (triggersKeyMap) {
+      setTriggersKeyMap(triggersKeyMap);
+    }
+  }, [triggersKeyMap]);
```

Comment:
It's better to use `triggerDetails` and `triggersKeyMap` directly. Otherwise, we will end up having race conditions when switching between different triggers.